common/src/main/java/us/myles/ViaVersion/handlers/CommonTransformer.java

```diff
@@ -0,0 +1,60 @@
+package us.myles.ViaVersion.handlers;
+
+import io.netty.buffer.ByteBuf;
+import us.myles.ViaVersion.api.PacketWrapper;
+import us.myles.ViaVersion.api.data.UserConnection;
+import us.myles.ViaVersion.api.type.Type;
+import us.myles.ViaVersion.exception.CancelException;
+import us.myles.ViaVersion.packets.Direction;
+import us.myles.ViaVersion.protocols.base.ProtocolInfo;
+
+import java.util.function.Function;
+
+public class CommonTransformer {
```

Comment:
These methods need javadocs as they are reusable by other platforms. 

I'd also argue that preServerboundCheck / preClientbound / transformClientbound / transformServerbound could be named differently.

My suggestions:
- willTransformPacket -> shouldTransformPacket
- preServerboundCheck -> checkIncomingPacket
- preClientbound -> checkOutgoingPacket
- transformClientbound -> transformOutgoing
- transformServerbound -> transformIncoming

Obviously these are suggestions, but the main point would be to keep consistency with wording across ViaVersion. The `pre` methods could probably be named better than what I suggested but they don't do the same thing which I guess is the issue.

I'd also argue that CommonTransformer shouldn't be static. If this was built into UserConnection or just created per connection we wouldn't have to pass the user in every time and it would be a lot more OOP.